nixos/doc/manual/development/systemd-state-management.section.md

```diff
@@ -0,0 +1,24 @@
+## State management {#sec-systemd-state-management}
+
+State refers to files and directories for config or running a service.
+A special section is dedicated at the end regarding databases.
+
+TODO: section on RuntimeDir, StateDir, WorkingDir
+
+### Database connection {#sec-systemd-state-management-database-connection}
+
+Most databases have 2 modes of connection, over TCP or over Unix socket. Connecting over a unix socket can provide 2 advantages
+- 30% performance improvement as packets don't need to be encoder over TCP
+- The default authentication mode postgres on unix sockets is just with the user name. This will remove the need to manage the password secret
```

Comment:
```suggestion
- The default authentication mode postgres on unix sockets relies on the client's uid provided by the kernel, rather than a password that you need to be care for. 
```

Reply:
Though it being based on uid, this may be tricky across uid namespaces and therefore containers. I vaguely recall a nixpkgs issue or stackoverflow question about this, but I couldn't find it.